ci: Add single concurrency to merge / revert

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #77424

Changes the merge / revert workflows to only run one job at a time

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

seemethere committed:
ci: Add single concurrency to merge / revert

Changes the merge / revert workflows to only run one job at a time

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

bigfootjon reviewed (approved):
LGTM, but since this might cause problems with merge-on-green I'm -1 on merging until that is resolved

seemethere committed:
Update on "ci: Add single concurrency to merge / revert"


Changes the merge / revert workflows to only run one job at a time

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on "ci: Add single concurrency to merge / revert"


Changes the merge / revert workflows to only run one job at a time

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere commented:
> LGTM, but since this might cause problems with merge-on-green I'm -1 on merging until that is resolved

Made a change to make it os that `merge-on-green` merges have their own concurrency queue

malfet reviewed (approved):
It wouldn't solve the problem, we really need to move merge and revert into one workflow

seemethere commented:
@pytorchbot merge on green
